pkg/armrpc/asyncoperation/worker/worker.go

```diff
@@ -251,27 +251,29 @@ func (w *AsyncRequestProcessWorker) runOperation(ctx context.Context, message *q
 
 		logger.Info("Start processing operation.")
 		result, err := asyncCtrl.Run(asyncReqCtx, asyncReq)
-
-		code := ""
-		if result.Error != nil {
-			code = result.Error.Code
+		// Update the result if an error is returned from the controller.
+		// Check that the result is empty to ensure we don't override it, it shouldn't happen.
+		// Controller should always either return non-empty error or non-empty result, but not both.
+		if err != nil && result.Error == nil {
+			armErr := extractError(err)
+			result.SetFailed(armErr, false)
 		}
 
-		logger.Info("Operation returned", "success", result.Error == nil, "code", code, "provisioningState", result.ProvisioningState(), "err", err)
-
 		// There are two cases when asyncReqCtx is canceled.
 		// 1. When the operation is timed out, w.completeOperation will be called in L186
 		// 2. When parent context is canceled or done, we need to requeue the operation to reprocess the request.
 		// Such cases should not call w.completeOperation.
 		if !errors.Is(asyncReqCtx.Err(), context.Canceled) {
-			if err != nil {
-				armErr := extractError(err)
-				result.SetFailed(armErr, false)
-				logger.Error(err, "Operation Failed")
-			}
-
 			w.completeOperation(ctx, message, result, asyncCtrl.StorageClient())
 		}
+
+		// Log the error if the operation failed.
+		if result.Error != nil {
+			logger.Error(result.Error, "Operation Failed")
+		} else {
+			logger.Info("Operation returned", "success", result.Error == nil, "provisioningState", result.ProvisioningState())
+		}
+
```

Comment:
I checked how logger.Error and logger.Info represent as a log finally.

Here is the example log `logger.Error` emits:

`{"severity":"error","timestamp":"2024-03-17T00:59:28.451Z","name":"radius.radiusasyncworker","caller":"worker/worker.go:270","message":"Operation Failed","serviceName":"radius","version":"edge","hostName":"applications-rp-59dfdc9c54-fjt56","resourceId":"/planes/radius/local/resourcegroups/kind-radius/providers/Applications.Core/containers/invalid-ctnr","operationId":"0cc83dfc-5dff-4441-be3d-d22ed0f727d1","operationType":"APPLICATIONS.CORE/CONTAINERS|PUT","dequeueCount":1,"traceId":"b82c1943e6b0b5aa4a64e20119a3144b","spanId":"9cab3ac40dc3ea1a","error":"code BadRequest: err linked \"/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/default/providers/applications.core/environments/env\" has invalid Applications.Core/applications resource type.","stacktrace":"github.com/radius-project/radius/pkg/armrpc/asyncoperation/worker.(*AsyncRequestProcessWorker).runOperation.func1\n\t/home/runner/work/radius/radius/pkg/armrpc/asyncoperation/worker/worker.go:270"}`

Here is the example log `logger.Info` emits:
`{"severity":"info","timestamp":"2024-03-17T00:59:28.625Z","name":"radius.radiusasyncworker","caller":"worker/worker.go:260","message":"Operation returned","serviceName":"radius","version":"edge","hostName":"applications-rp-59dfdc9c54-fjt56","resourceId":"/planes/radius/local/resourcegroups/kind-radius/providers/Applications.Core/gateways/invalid-gtwy","operationId":"1ff4da3e-7043-4869-a862-311f10ae7ecd","operationType":"APPLICATIONS.CORE/GATEWAYS|PUT","dequeueCount":1,"traceId":"b82c1943e6b0b5aa4a64e20119a3144b","spanId":"535fe18ef22aff07","success":true,"code":"","provisioningState":"Succeeded","err":"code BadRequest: err application ID \"not_an_id\" for the resource \"/planes/radius/local/resourcegroups/kind-radius/providers/Applications.Core/gateways/invalid-gtwy\" is not a valid id. Error: 'not_an_id' is not a valid resource id"}`

One of the benefits to use logger.Error is adding `stacktrace` attribute. If we want to create query for displaying chart, the query might be complicated. 
It seems like using only `logger.Info` might be easier for query the logs because we can filter only `Operation returned` message with success attribute and error attribute (for the failed operation)
I initially said I did not have the strong opinion, but when I look at the actual log output, using only logger.Info might be easier to make the query for chart.

Reply:
Discussed offline and updated everything to be logged as info.